test/integration/cni/pod_traffic_across_az_test.go

```diff
@@ -0,0 +1,304 @@
+		err           error
+		testDaemonSet *v1.DaemonSet
+
+		// Map of Pods placed on primary/secondary ENI IP
+		interfaceToPodList common.InterfaceTypeToPodList
+
+		azName string
+
+		// Map of AZ name to Pod
+		azToPod map[string]coreV1.Pod
+	)
+
+	JustBeforeEach(func() {
+		serverContainer := manifest.
+			NewCurlContainer().
+			Command([]string{
+				"sleep",
+				"3600",
+			}).
+			Build()
+
+		By("creating a server DaemonSet on primary node")
+
+		testDaemonSet = manifest.
+			NewDefaultDaemonsetBuilder().
+			Container(serverContainer).
+			PodLabel("role", "az-test").
+			Name("api-server-connectivity-daemonset").
+			Build()
+
+		_, err = f.K8sResourceManagers.DaemonSetManager().CreateAndWaitTillDaemonSetIsReady(testDaemonSet, utils.DefaultDeploymentReadyTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		By(fmt.Sprintf("getting the node with the node label key %s and value %s",
+			f.Options.NgNameLabelKey, f.Options.NgNameLabelVal))
+
+		nodes, err := f.K8sResourceManagers.NodeManager().GetNodes(f.Options.NgNameLabelKey, f.Options.NgNameLabelVal)
+		Expect(err).ToNot(HaveOccurred())
+
+		azToPod = make(map[string]coreV1.Pod)
+
+		for i := range nodes.Items {
+			// node label key "topology.kubernetes.io/zone" is well known label populated by cloud controller manager
+			// guaranteed to be present and represent the AZ name
+			// Ref https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone
+			azName = nodes.Items[i].ObjectMeta.Labels["topology.kubernetes.io/zone"]
+			interfaceToPodList = common.GetPodsOnPrimaryAndSecondaryInterface(nodes.Items[i], "role", "az-test", f)
+			if len(interfaceToPodList.PodsOnSecondaryENI) > 0 {
+				azToPod[azName] = interfaceToPodList.PodsOnSecondaryENI[0]
+			}
+			if len(interfaceToPodList.PodsOnPrimaryENI) > 0 {
+				azToPod[azName] = interfaceToPodList.PodsOnPrimaryENI[0]
+			}
+		}
+	})
+
+	JustAfterEach(func() {
+		By("Deleting the Daemonset.")
+		err = f.K8sResourceManagers.DaemonSetManager().DeleteAndWaitTillDaemonSetDeleted(testDaemonSet, utils.DefaultDeploymentReadyTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+	})
+
+	Context("While testing API Server Connectivity", func() {
+		It("Should connect to the API Server", func() {
+			// Standard paths for SA token, CA cert and API Server URL
+			token_path := "/var/run/secrets/kubernetes.io/serviceaccount/token"
+			cacert := "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+			api_server_url := "https://kubernetes.default.svc/api"
+
+			for az := range azToPod {
+				fmt.Printf("Testing API Server Connectivity from AZ %s \n", az)
+				sa_token := []string{"cat", token_path}
+				token_value, _, err := RunCommandOnPod(azToPod[az], sa_token)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to get SA token for pod in %s", az))
+				bearer := fmt.Sprintf("Authorization: Bearer %s", token_value)
+				test_api_server_connectivity := []string{"curl", "--cacert", cacert, "--header", bearer, "-X", "GET",
```

Comment:
We can avoid this and probably just reach out to `/healthz` endpoint but I think this is more complete ..

Reply:
We should also test the NLB path to API Server - Above tests the X-ENI path.

Reply:
also, am wondering if we should repeat this few times to make sure our request from this node/pod will access both the API Server endpoints (or)...we manually curl to individual API server endpoints